### PR TITLE
adding guard clauses to endpoints provider for non-local condition

### PR DESCRIPTION
**Pull request by timmyreilly**
Signed-off-by: timmyreilly <tireilly@microsoft.com>

Handling non-local services in endpoints provider with guard clauses. 
#3456 and #3441 

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [X] |
| Documentation              | [ ] |
| Install                    | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Ingress                    | [ ] |
| Egress                     | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| SMI Policy                 | [ ] |
| Sidecar Injection          | [ ] |
| Security                   | [ ] |
| Upgrade                    | [ ] |
| Tests                      | [ ] |
| CI System                  | [ ] |
| Demo                       | [ ] |
| Performance                | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?
No
1. Is this a breaking change?
No

### CODE REVIEW

File: pkg/endpoint/providers/kube/client.go [hunk 1]
```diff
@@ -36,6 +36,11 @@ func (c Client) ListEndpointsForService(svc service.MeshService) []endpoint.Endp
 	log.Trace().Msgf("[%s] Getting Endpoints for service %s on Kubernetes", c.providerIdent, svc)
 	var endpoints []endpoint.Endpoint
 
+	if !svc.IsLocal() {
+		log.Trace().Msg("Not a local service and we can't handle that scenario yet.")
```
Comment (shashankram) [inline pkg/endpoint/providers/kube/client.go:40]:
```suggestion
		log.Warn().Msgf("%s is not a local service, cannot retrieve endpoints", svc)
```

File: pkg/service/types.go [hunk 2]
```diff
@@ -18,6 +18,10 @@ type MeshService struct {
 	Name string
 }
 
+func (ms MeshService) IsLocal() bool {
+	return true
```
Comment (snehachhabria) [inline pkg/service/types.go:22]: Why is defaulting to true, shouldn't there be some logic to figure it out from mesh service

Comment (timmyreilly) [reply pkg/service/types.go:22]: Placeholder until we implement TODO: Implement this. Do an exact match for local on .ClusterDomain

Comment (steeling) [reply pkg/service/types.go:22]: tim can you add a TODO comment on this?

**Pull request closed without merging**